pkg/manager/member/tikv_upgrader.go

```diff
@@ -14,26 +14,162 @@
 package member
 
 import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/tidb-operator/pkg/apis/pingcap.com/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
 	apps "k8s.io/api/apps/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
 )
 
-type tikvUpgrader struct{}
+const (
+	EvictLeaderBeginTime = "evictLeaderBeginTime"
+	EvictLeaderTimeOut   = 3 * time.Minute
```

Comment:
Does 3 minutes enough?

Reply:
This method does not guarantee that all region leaders can be transferred. Setting timeout is just to avoid long-term blocking, and it is not necessary to set it too long. I believe that 3 minutes is enough for most of the region leaders to transfer.